tests/lab1_test.cpp

```diff
@@ -0,0 +1,92 @@
+#include <gtest/gtest.h>
+
+#include <array>
+#include <filesystem>
+#include <fstream>
+#include <memory>
+
+namespace fs = std::filesystem;
+
+extern "C" {
+#include "lab1.h"
+}
+
+TEST(FirstLabTests, SimpleTest) {
+    const char *childPath = getenv("PATH_TO_CHILD");
+    ASSERT_TRUE(childPath);
+
+    const std::string fileWithInput = "input.txt";
+    const std::string fileWithOutput1 = "output1.txt";
+    const std::string fileWithOutput2 = "output2.txt";
+
+    constexpr int outputSize1 = 5;
+    constexpr int outputSize2 = 3;
+    constexpr int inputSize = outputSize1 + outputSize2;
+
+    std::array<std::string, inputSize> input = {
+        "rev",
+        "revvvvv",
+        "162te16782r18223r2",
+        "",
+        "r",
+        "pqwrpqlwfpqwoglqwpglqpgwpqw.,g;q.wg;q.wg;w.qg;.w",
+        "12345678900",
+        "1234567890",
+    };
+
+    std::array<std::string, outputSize1> expectedOutput1 = {
+        "ver",
+        "vvvvver",
+        "",
+        "r",
+        "0987654321",
+    };
+
+    std::array<std::string, outputSize2> expectedOutput2 = {
+        "2r32281r28761et261",
+        "w.;gq.w;gw.q;gw.q;g,.wqpwgpqlgpwqlgowqpfwlqprwqp",
+        "00987654321",
+    };
+
+    {
+        auto inFile = std::ofstream(fileWithInput);
+        inFile << fileWithOutput1 << '\n'
+               << fileWithOutput2 << '\n';
+        for (const auto &line : input) {
+            inFile << line << '\n';
+        }
+    }
+
+    auto deleter = [](FILE *file) {
+        fclose(file);
+    };
+
+    std::unique_ptr<FILE, decltype(deleter)> inFile(fopen(fileWithInput.c_str(), "r"), deleter);
+
+    ASSERT_TRUE(ParentRoutine(childPath, inFile.get()) == 0);
```

Comment:
```ASSERT_EQ(ParentRoutine(childPath, inFile.get()), 0)```

Reply:
fixed in 16c8a67ba6e4661b8c59561c592ba9456b6312ec